### ISSUE TITLE
False-positive type error in text mode when pipeline operator is used

### ISSUE THREAD

**Issue by zeptometer** [gfngfn/SATySFi#257]
When compiling this `hoge.saty` file with text mode,
```
() |> (fun () -> `a`)
```

I got the following type error.
```
$ satysfi --text-mode "text" test/hoge.saty -o main.txt
 ---- ---- ---- ----
  target file: 'main.txt'
  dump file: 'hoge.satysfi-aux' (already exists)
  parsing 'hoge.saty' ...
 ---- ---- ---- ----
  type checking 'hoge.saty' ...
  type check passed. (string)
! [Type Error] file 'hoge.saty' is not a file for generating text; it is of type
      string
```

It does not happen when I omit the pipeline operator.
```
(fun () -> `a`) ()
```

```
$ satysfi --text-mode "text" test/hoge.saty -o main.txt
 ---- ---- ---- ----
  target file: 'main.txt'
  dump file: 'hoge.satysfi-aux' (already exists)
  parsing 'hoge.saty' ...
 ---- ---- ---- ----
  type checking 'hoge.saty' ...
  type check passed. (string)
  preprocessing 'hoge.saty' ...
 ---- ---- ---- ----
  evaluating texts ...
  evaluation done.
  all cross references were solved.
 ---- ---- ---- ----
  output written on 'main.txt'.
```

**Comment by zeptometer** [comment]
SATySFi version is 0.0.5.

**Comment by gfngfn** [comment]
Thank you for reporting! The strange behavior seems fixed at a9d762b.  Its cause can be seen in d5884a2.

Example:

```console
$ cat hoge.saty
() |> (fun () -> `a`)
$ satysfi --text-mode "text" hoge.saty -o main.txt      
 ---- ---- ---- ----
  target file: 'main.txt'
  dump file: 'hoge.satysfi-aux' (will be created)
  parsing 'hoge.saty' ...
 ---- ---- ---- ----
  type checking 'hoge.saty' ...
  type check passed. (string)
  preprocessing 'hoge.saty' ...
 ---- ---- ---- ----
  evaluating texts ...
  evaluation done.
  all cross references were solved.
 ---- ---- ---- ----
  output written on 'main.txt'.
$ cat main.txt                    
a
```

**Comment by zeptometer** [comment]
Nice, thank you for fixing it!
